orderbook/src/database/orders.rs

```diff
@@ -0,0 +1,208 @@
+            OrderKind::Sell => Self::Sell,
+        }
+    }
+
+    fn into(self) -> OrderKind {
+        match self {
+            Self::Buy => OrderKind::Buy,
+            Self::Sell => OrderKind::Sell,
+        }
+    }
+}
+
+impl Database {
+    // TODO: Errors if order uid already exists. We might want to have different behavior like
+    // indicating this in the return value or simply allowing it to happen.
+    pub async fn insert_order(&self, order: &Order) -> Result<()> {
+        const QUERY: &str = "\
+            INSERT INTO orders (
+                uid, owner, creation_timestamp, sell_token, buy_token, sell_amount, buy_amount, \
+                valid_to, app_data, fee_amount, kind, partially_fillable, signature) \
+            VALUES ( \
+                $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13);";
+        sqlx::query(QUERY)
+            .bind(order.order_meta_data.uid.0.as_ref())
+            .bind(order.order_meta_data.owner.as_bytes())
+            .bind(order.order_meta_data.creation_date)
+            .bind(order.order_creation.sell_token.as_bytes())
+            .bind(order.order_creation.buy_token.as_bytes())
+            .bind(u256_to_big_decimal(&order.order_creation.sell_amount))
+            .bind(u256_to_big_decimal(&order.order_creation.buy_amount))
+            .bind(order.order_creation.valid_to)
+            .bind(order.order_creation.app_data)
+            .bind(u256_to_big_decimal(&order.order_creation.fee_amount))
+            .bind(DbOrderKind::from(order.order_creation.kind))
+            .bind(order.order_creation.partially_fillable)
+            .bind(order.order_creation.signature.to_bytes().as_ref())
+            .execute(&self.pool)
+            .await
+            .context("insert_order failed")
+            .map(|_| ())
+    }
+
+    // TODO: add filters: not_fully_executed, owner, sell_token, buy_token
+    pub fn orders(&self, min_valid_to: u32) -> impl Stream<Item = Result<Order>> + '_ {
+        // TODO: adapt query to filter fully executed orders immediately
+        const QUERY: &str = "\
+            SELECT \
+                o.uid, o.owner, o.creation_timestamp, o.sell_token, o.buy_token, o.sell_amount, \
+                o.buy_amount, o.valid_to, o.app_data, o.fee_amount, o.kind, o.partially_fillable, \
+                o.signature, \
+                COALESCE(SUM(t.sell_amount), 0) AS sum_sell, \
+                COALESCE(SUM(t.buy_amount), 0) AS sum_buy, \
+                COALESCE(SUM(t.fee_amount), 0) AS sum_fee \
+            FROM orders o LEFT OUTER JOIN trades t ON o.uid = t.order_uid
+            WHERE o.valid_to >= $1 \
+            GROUP BY o.uid;";
+        // To get only not fully executed orders we probably want to use something like
+        // `HAVING sum_sell < orders.sell_amount` but still need to find the best way to make this
+        // pick the correct column based on order type.
+
+        #[derive(sqlx::FromRow)]
+        struct Row {
```

Comment:
nit: Maybe moving row outside of this function and implementing TryInto<Order> makes the second half of this function smaller and easier to read (assuming it will become more complex when we add filtering).